### PR TITLE
feat: Retry with reloaded credentials on cloud error

### PR DESCRIPTION
**Pull request by nameexhaustion**
ref https://github.com/pola-rs/polars/issues/19933

### CODE REVIEW

File: crates/polars-io/src/cloud/glob.rs [hunk 1]
```diff
@@ -74,13 +75,13 @@ pub(crate) fn extract_prefix_expansion(url: &str) -> PolarsResult<(String, Optio
 #[derive(PartialEq, Debug, Default)]
 pub struct CloudLocation {
     /// The scheme (s3, ...).
-    pub scheme: String,
+    pub scheme: PlSmallStr,
```
Comment (nameexhaustion) [inline crates/polars-io/src/cloud/glob.rs:78]: drive-by `PlSmallStr` as these tend to be small

**Review by ritchie46 — APPROVED** [review]
